src/binary/type_code.rs

```diff
@@ -88,3 +96,28 @@ impl IonTypeCode {
         Ok(ion_type_code)
     }
 }
+
+impl IonTypeCode {
+    /// Constant function to convert an [`IonTypeCode`] into a `u8`.
+    pub const fn to_u8(self) -> u8 {
```

Comment:
Just curious: why write a `to_u8` function instead of providing an `impl From<IonTypeCode> for u8`?

I thought it might be because you couldn't impl a trait for a type outside the current crate, but [that seems to have changed (at least for the `From` trait?) in Rust 1.41](https://doc.rust-lang.org/std/convert/trait.Into.html#implementing-into-for-conversions-to-external-types-in-old-versions-of-rust). (See working playground example [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=93b7f1923030d832cd65c33c6a237f0a)).

`const` could be buying us a minor performance benefit; is that the motivation?

Reply:
Yeah, it's all about const. The previous commit message said something like "this is not From because of const" and Almann suggested I rewrite it to the above.

I haven't benchmarked `const`. I just did it like this because it was the weekend and I was having fun!